src/arrayMethodJoin.js

```diff
@@ -4,8 +4,22 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let joined = '';
+
+    joined += this[0] !== undefined && this[0] !== null
+      ? this[0]
+      : '';
```

Comment:
I don't understand why you set the same condition twice in the code.
You can use it once in the middle of the cycle